Test_recorder test complains about D800 character on CygWin with py27

On cygwin64_py27, with minimum package levels, on master branch.

See this Appveyor run: https://ci.appveyor.com/project/andy-maier/pywbem/builds/32193355/job/l287lvjhmq8wl1xy#L2220:

```
=================================== FAILURES ===================================
 test_TestClientRecorder_record[CreateInstance with properties of all types-CreateInstance-op_kwargs2-op_result2-None-exp_yaml_items2] 
desc = 'CreateInstance with properties of all types', op_name = 'CreateInstance'
op_kwargs = {'NewInstance': CIMInstance(classname='CIM_Foo', path=None, properties=NocaseDict({'B1': CIMPr...e, qualifiers=NocaseDict({}))}), property_list=None, qualifiers=NocaseDict({})), 'namespace': 'cim/blah'}
op_result = CIMInstanceName(classname='C1', keybindings=NocaseDict({}), namespace=None, host=None)
op_exc = None
exp_yaml_items = {'name': 'CreateInstance', 'pywbem_request': {'creds': ['username', 'password'], 'debug': False, 'namespace': 'root/ci...'pywbem_response': {'result': {'classname': 'C1', 'host': None, 'keybindings': OrderedDict(), 'namespace': None, ...}}}
test_client_recorder = <pywbem._recorder.TestClientRecorder object at 0x6fff27d5f250>
    @pytest.mark.parametrize(
        "desc, op_name, op_kwargs, op_result, op_exc, exp_yaml_items",
        TESTCASES_TESTCLIENTRECORDER_RECORD)
    def test_TestClientRecorder_record(
            desc, op_name, op_kwargs, op_result, op_exc, exp_yaml_items,
            test_client_recorder):
        # pylint: disable=redefined-outer-name
        """
        Record a single operation using the test client recorder and verify the
        generated YAML file.
        """
    
        # The code to be tested
        if op_name.startswith('Open') or op_name.startswith('Pull'):
            test_client_recorder.reset(pull_op=True)
        test_client_recorder.stage_pywbem_args(method=op_name, **op_kwargs)
        test_client_recorder.stage_pywbem_result(op_result, op_exc)
        test_client_recorder.record_staged()
    
        if DEBUG_TEST_YAML_FILE:
            print("\nDebug: Test client YAML file for testcase: {}".
                  format(desc))
            if sys.platform == 'win32':
                os.system('type {}'.format(TEST_YAML_FILE))
            else:
                os.system('cat {}'.format(TEST_YAML_FILE))
    
        # Verify the generated test client YAML file
>       test_yaml = load_recorder_yaml_file()
tests/unittest/pywbem/test_recorder.py:1538: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unittest/pywbem/test_recorder.py:769: in load_recorder_yaml_file
    fp, Loader=yamlloader.ordereddict.CSafeLoader)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/__init__.py:114: in load
    return loader.get_single_data()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/constructor.py:43: in get_single_data
    node = self.get_single_node()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:36: in get_single_node
    document = self.compose_document()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:55: in compose_document
    node = self.compose_node(None, None)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:82: in compose_node
    node = self.compose_sequence_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:111: in compose_sequence_node
    node.value.append(self.compose_node(node, index))
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:84: in compose_node
    node = self.compose_mapping_node(anchor)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:133: in compose_mapping_node
    item_value = self.compose_node(node, item_key)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/composer.py:64: in compose_node
    if self.check_event(AliasEvent):
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/parser.py:98: in check_event
    self.current_event = self.state()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/parser.py:449: in parse_block_mapping_value
    if not self.check_token(KeyToken, ValueToken, BlockEndToken):
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/scanner.py:116: in check_token
    self.fetch_more_tokens()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/scanner.py:255: in fetch_more_tokens
    return self.fetch_plain()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/scanner.py:679: in fetch_plain
    self.tokens.append(self.scan_plain())
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/scanner.py:1311: in scan_plain
    spaces = self.scan_plain_spaces(indent, start_mark)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/scanner.py:1329: in scan_plain_spaces
    line_break = self.scan_line_break()
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/scanner.py:1436: in scan_line_break
    if self.prefix(2) == u'\r\n':
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/reader.py:98: in prefix
    self.update(length)
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/reader.py:174: in update
    self.check_printable(data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <yamlloader.ordereddict.loaders.CSafeLoader object at 0x6fff2864e0d0>
data = '                        reference_class: null
                        embedded_object: null
                        i...lue: A𐅂B
                        type: string
                        reference_class: null
                        e'
    def check_printable(self, data):
        match = self.NON_PRINTABLE.search(data)
        if match:
            character = match.group()
            position = self.index+(len(self.buffer)-self.pointer)+match.start()
            raise ReaderError(self.name, position, ord(character),
>                   'unicode', "special characters are not allowed")
E           ReaderError: unacceptable character #xd800: special characters are not allowed
E             in "/cygdrive/c/projects/pywbem/tests/unittest/pywbem/test_recorder.yaml", position 2960
.tox/cygwin64_py27/lib/python2.7/site-packages/yaml/reader.py:149: ReaderError
```

--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -413,6 +413,10 @@ Released: not yet
   versions that support a particular Python versions beyond that.
   This only affects development of pywbem. (See issue #2174)
 
+* Increased the version of 'PyYAML' from 5.1 to 5.3 on Python 2.7, to pick
+  up a fix for dealing with Unicode characters above U+FFFF in narrow Python
+  builds. (See issue #2182)
+
 **Enhancements:**
 
 * Changed GetCentralInstances methodology in WBEMServer.get_central_instances()

--- a/makefile
+++ b/makefile
@@ -386,6 +386,7 @@ platform:
 	@echo "Python command location: $(shell $(WHICH) $(PYTHON_CMD))"
 	@echo "Python version: $(python_version)"
 	@$(PYTHON_CMD) tools/python_bitsize.py
+	@$(PYTHON_CMD) tools/python_unicodesize.py
 	@echo "Pip command name: $(PIP_CMD)"
 	@echo "Pip command location: $(shell $(WHICH) $(PIP_CMD))"
 	@echo "$(package_name) package version: $(package_version)"

--- a/minimum-constraints.txt
+++ b/minimum-constraints.txt
@@ -88,7 +88,8 @@ wheel==0.33.5; python_version >= '3.8'
 mock==2.0.0
 ordereddict==1.1
 ply==3.10
-PyYAML==5.1
+PyYAML==5.3; python_version == '2.7'  # PyYAML 5.3 fixes narrow build error
+PyYAML==5.1; python_version >= '3.4'
 six==1.10.0; python_version < '3.8'
 six==1.12.0; python_version >= '3.8'  # required by virtualenv==20.0.0
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -12,8 +12,8 @@
 mock>=2.0.0,<4.0.0; python_version < '3.6'
 mock>=2.0.0; python_version >= '3.6'
 ply>=3.10
-# PyYAML 5.3 has removed support for Python 3.4
-PyYAML>=5.1; python_version == '2.7'
+# PyYAML 5.3 has removed support for Python 3.4; fixes narrow build error
+PyYAML>=5.3; python_version == '2.7'
 PyYAML>=5.1,<5.3; python_version == '3.4'
 PyYAML>=5.1; python_version > '3.4'
 # virtualenv 20.0.0 (required on py3.8+) requires six>=0.12.0

--- a/tools/python_unicodesize.py
+++ b/tools/python_unicodesize.py
@@ -1,0 +1,34 @@
+#!/usr/bin/env python
+
+import sys
+
+# Method 1: Test whether s.decode() returns surrogate pair or single char
+s = b'\\U00010142'
+c = s.decode('unicode-escape')
+if len(c) == 1:
+    size1 = 'wide'
+elif len(c) == 2:
+    size1 = 'narrow'
+else:
+    raise AssertionError(
+        "python_unicodesize.py: Length of Unicode character {0} returned from "
+        "s.decode() must be 1 or 2 but is: {1}".format(s, len(c)))
+
+# Method 2: Test whether unichr() fails
+try:
+    if sys.version_info[0] == 2:
+        unichr(65858)  # U+10142
+    else:
+        chr(65858)  # U+10142
+except ValueError:
+    size2 = 'narrow'
+else:
+    size2 = 'wide'
+
+if size1 != size2:
+    raise AssertionError(
+        "python_unicodesize.py: Different results for unicode size: "
+        "s.decode() method: {0}, unichr() method: {1}".
+        format(size1, size2))
+
+print("Unicode size of Python environment: {0}".format(size1))

